src/js/libcs/OpImageDrawing.js

```diff
@@ -347,12 +347,102 @@ evaluator.allimages$0 = function() {
+                width = val.value[0].value.real;
+                heightorratio = val.value[1].value.real;
+            } else return nada;
+
+            constraints = {
+                video: {
+                    width: width,
+                    advanced: [{
+                        width: {
+                            max: width, //ideal or exact seem not be supported by Chrome yet
+                            min: width
+                        }
+                    }, {
+                        width: {
+                            ideal: width
+                        }
+                    }]
+                },
+                audio: false
+            };
+            if (heightorratio) {
+                if (heightorratio < 10 || !Number.isInteger(heightorratio)) {
+                    constraints.video.aspectRatio = heightorratio;
+                    constraints.video.advanced[0].aspectRatio = {
+                        min: heightorratio,
+                        max: heightorratio
+                    };
+                    constraints.video.advanced[1].aspectRatio = {
+                        ideal: heightorratio,
+                    };
+                } else {
+                    constraints.video.height = heightorratio;
+                    constraints.video.advanced[0].height = {
+                        min: heightorratio,
+                        max: heightorratio
+                    };
+                    constraints.video.advanced[1].height = {
+                        ideal: heightorratio,
+                    };
+                }
+            }
+        }
+    }
+    if (maximal) {
+        // As per https://bugs.chromium.org/p/chromium/issues/detail?id=543997#c47,
+        // Chrome 54 doesn't actually honor ideal constraints yet, so we need
+        // to explicitely list some common widths to control resolution selection.
+        constraints = [320, 640, 1024, 1280, 1920, 2560];
+        constraints = constraints.map(function(w) {
+            return {
+                width: {
+                    min: w
+                }
+            };
+        });
+        // We'd like to also minimize aspect ratio i.e. maximize height for a given
+        // width, but Chrome again appears to have a problem with this. See also
+        // https://bugs.chromium.org/p/chromium/issues/detail?id=657145
+        if (false) {
+            constraints = constraints.concat([1.34, 1.59, 1.78].map(function(a) {
+                return {
+                    aspectRatio: {
+                        max: a
+                    }
+                };
+            }));
+        }
+        constraints = {
+            video: {
+                width: 16000, // ideal dimensions, will
+                height: 9000, // prefer big resolutions
+                advanced: constraints
+            },
+            audio: false
+        };
+    }
+
+    console.log(constraints);
```

Comment:
Do we want to keep this? Perhaps, for the time being. In the long run it should go away, though.